File: Server/src/Server.Application/ViewModels/UserProfileViewModel.cs
namespace Server.Application.ViewModels;

public class UserProfileViewModel
{
    public required string UserName { get; set; }
    public string? UserImageUrl { get; set; }
    public List<BookViewModel>? UserLastReadBooks { get; set; }
    public string? UserLocation { get; set; }
    public string? AboutMe { get; set; }
    public int? ReadBooksCount { get; set; }
    public int FollowersCount { get; set; }
    public bool IsCritic { get; set; }
    public DateTime RegisteredAt { get; set; }
    public bool? FollowedByMe { get; set; }
}

File: client/src/models/user/UserProfileViewModel.ts
import { z } from 'zod';
import BookViewModel from '../BookViewModel';

const UserProfileViewModel = z.object({
  userName: z.string(),
  userImageUrl: z.string().nullable(),
  userLocation: z.string().nullable(),
  userLastReadBooks: BookViewModel.array(),
  aboutMe: z.string().nullable(),
  readBooksCount: z.number().nullable(),
  registeredAt: z.string(),
  isCritic: z.boolean(),
  followedByMe: z.boolean().nullable(),
  followersCount: z.number(),
});

export default UserProfileViewModel;
export type UserProfileViewModelType = z.infer<typeof UserProfileViewModel>;
